src/Node/HTTP.purs

```diff
@@ -51,6 +77,194 @@ foreign import data HTTP :: Effect
 -- | Create a HTTP server, given a function to be executed when a request is received.
 foreign import createServer :: forall eff. (Request -> Response -> Eff (http :: HTTP | eff) Unit) -> Eff (http :: HTTP | eff) Server
 
+-- | The type of HTTPS server options
+data SSLOptions
+
+-- | Abort the connection if the SSL/TLS handshake does not finish in the
+-- | specified number of milliseconds. Defaults to 120 seconds. A
+-- | 'tlsClientError' is emitted on the tls.Server object whenever a handshake
+-- | times out.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+handshakeTimeout :: Option SSLOptions Int
+handshakeTimeout = opt "handshakeTimeout"
+
+-- | If true the server will request a certificate from clients that connect and
+-- | attempt to verify that certificate. Defaults to false.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+requestCert :: Option SSLOptions Boolean
+requestCert = opt "requestCert"
+
+-- | If not false the server will reject any connection which is not authorized
+-- | with the list of supplied CAs. This option only has an effect if
+-- | requestCert is true. Defaults to true.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+rejectUnauthorized :: Option SSLOptions Boolean
+rejectUnauthorized = opt "rejectUnauthorized"
+
+-- | An array of strings, Buffers or Uint8Arrays, or a single Buffer or
+-- | Uint8Array containing supported NPN protocols. Buffers should have the
+-- | format [len][name][len][name]... e.g. 0x05hello0x05world, where the first
+-- | byte is the length of the next protocol name. Passing an array is usually
+-- | much simpler, e.g. ['hello', 'world']. (Protocols should be ordered by
+-- | their priority.)
+-- | The type variable t should be a string[], Buffer[], Uint8Array[], Buffer,
+-- | or Uint8Array.
+-- | See the [node docs](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener)
+npnProtocols :: forall t. Option SSLOptions t
```

Comment:
I had originally tried to do the first solution, but the problem is you end up with something like this:

```purescript
data NPNProtocols = ProtocolString Strings | ...
npnProtocols :: Option SSLOptions NPNProtocols
npnProtocols = opt "NPNProtocols"

...

npnProtocolOption :: Options SSLOptions
npnProtocolOption = npnProtocols := ProtocolString "..."
```

But now `npnProtocolOption` contains, for the option `npnProtocols`, a value of type `ProtocolString String`, not a value of type `String`.  This does not pass down to Node properly.

I _could_ unwrap this in `createServerSImpl` by doing something like `options.key = options.key.value0;`, but I don't know how reliable (or conventional) that is...

I agree that the typeclass approach isn't great as the valid types are dictated by the underlying Node API and should explicitly _not_ be extensible.

I'm not thrilled with using `foreign import data` as that would require a fairly different API for working with these options than other options.  Also since there are a whole bunch of different options to this API that accept a few different types, this would lead to a really large surface area.